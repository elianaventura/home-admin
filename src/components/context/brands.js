import { useState, createContext } from "react";

const BrandsContext = createContext({});

export function BrandsProvider({ children, value }) {
  const [brands, setBrands] = useState(value || []);

  return (
    <BrandsContext.Provider value={{ brands, setBrands }}>
      {children}    
    </BrandsContext.Provider>
  )
};

export default BrandsContext;
