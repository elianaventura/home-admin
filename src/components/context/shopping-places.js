import { useState, createContext } from "react";

const ShoppingPlacesContext = createContext({});

export function ShoppingPlacesProvider({ children, value }) {
  const [places, setShoppingPlaces] = useState(value || []);

  return (
    <ShoppingPlacesContext.Provider value={{ places, setShoppingPlaces }}>
      {children}    
    </ShoppingPlacesContext.Provider>
  )
};

export default ShoppingPlacesContext;
