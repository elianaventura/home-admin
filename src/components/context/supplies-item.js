import { useState, createContext } from "react";

const SuppliesItemContext = createContext({});

export function SuppliesItemProvider({ children, value }) {
  const [unit, setUnit] = useState(value);

  return (
    <SuppliesItemContext.Provider value={{ unit, setUnit }}>
      {children}    
    </SuppliesItemContext.Provider>
  )
};

export default SuppliesItemContext;
