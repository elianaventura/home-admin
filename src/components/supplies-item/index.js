"use client"

export default function SuppliesItem() {
  return (
    <div>
      ITEM DE SUPERMERCADO
    </div>
  );
};
