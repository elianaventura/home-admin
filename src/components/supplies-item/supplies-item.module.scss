.container {
  display: block;
}

.item-edit-icon {
  margin-left: auto;
}
