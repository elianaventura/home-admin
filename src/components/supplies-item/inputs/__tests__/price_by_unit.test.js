import { render } from "@testing-library/react";
import PriceByUnit from "../price_by_unit";
import { SuppliesItemProvider } from "@/components/context/supplies-item";
import userEvent from '@testing-library/user-event';
import { screen } from '@testing-library/dom';

const props = {
  field_name: 'price_per_unit',
  title: 'Precio',
  amount: 100,
  currency: {
    currency_selected_label: 'Moneda',
    options: [
      {
        id: 'ARS',
        selected: true,
        display: '$',
      },
      {
        id: 'USD',
        selected: false,
        display: 'USD',
      },
    ],
  },
  handleFieldChange: jest.fn(),
};

const unit = {
  id: 'KG',
  selected: true,
  display: {
    singular: 'kg',
    plural: 'kg',
  },
};

describe('First', () => {
  it('renders passes snapshot', () => {
    const { container } = render(
      <SuppliesItemProvider value={unit}>
        <PriceByUnit {...props} />
      </SuppliesItemProvider>
    );
 
    expect(container).toMatchSnapshot();
  });

  it('currency default selected currency is rendered', () => {
    const { getByText } = render(
      <SuppliesItemProvider value={unit}>
        <PriceByUnit {...props} />
      </SuppliesItemProvider>
    );
 
    expect(getByText('$')).toBeInTheDocument();
  });

  it('should change currency', async () => {
    const user = userEvent.setup();
    const { getByText } = render(
      <SuppliesItemProvider value={unit}>
        <PriceByUnit {...props} />
      </SuppliesItemProvider>
    );
 
    const selector = getByText('$');
    expect(screen.queryByText('USD')).toBeNull();

    await user.click(selector);

    const newOption = getByText('USD');
    await user.click(newOption);

    expect(getByText('USD')).toBeInTheDocument();
    expect(props.handleFieldChange).toHaveBeenCalledWith(
      props.field_name,
      {
        amount: props.amount,
        currency: 'USD',
        unit: 'KG',
      },
    );
  });

  it('should change price amount', async () => {
    const user = userEvent.setup();
    const { getByTestId } = render(
      <SuppliesItemProvider value={unit}>
        <PriceByUnit {...props} />
      </SuppliesItemProvider>
    );
 
    const amountInput = getByTestId('price-amount-input');
    await userEvent.clear(amountInput);
    await userEvent.type(amountInput, '1');

    expect(props.handleFieldChange).toHaveBeenCalledWith(
      props.field_name,
      {
        amount: '1',
        currency: 'ARS',
        unit: 'KG',
      },
    );
  });
});
