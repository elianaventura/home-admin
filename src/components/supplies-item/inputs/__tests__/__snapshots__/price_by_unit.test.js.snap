// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`First renders passes snapshot 1`] = `
<div>
  <div>
    <span>
      Precio
    </span>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary currency-select css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="currency-select-label currency-select"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="currency-select"
        role="button"
        tabindex="0"
      >
        $
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value="ARS"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-ihdtdm"
        >
          <span
            class="notranslate"
          >
            â€‹
          </span>
        </legend>
      </fieldset>
    </div>
    <input
      class="amount-input"
      data-testid="price-amount-input"
      value="100"
    />
  </div>
</div>
`;
